job/server/src/main/java/alluxio/job/transform/format/csv/CsvReader.java

```diff
@@ -70,23 +86,10 @@ private static InputStream open(FileSystem fs, Path path) throws IOException {
    * @throws IOException when failed to create the reader
    */
   public static CsvReader create(AlluxioURI uri) throws IOException {
-    CSVProperties props = new CSVProperties.Builder()
-        .hasHeader()
-        .build();
     Path inputPath = new Path(uri.getScheme(), uri.getAuthority().toString(), uri.getPath());
     Configuration conf = ReadWriterUtils.readNoCacheConf();
     FileSystem fs = inputPath.getFileSystem(conf);
-    Schema schema;
-    try (InputStream inputStream = open(fs, inputPath)) {
-      String schemaName = inputPath.getName();
-      if (schemaName.contains(".")) {
-        schemaName = schemaName.substring(0, schemaName.indexOf("."));
-      }
-      schema = AvroCSV.inferSchema(schemaName, inputStream, props);
-    }
-    AvroCSVReader<Record> reader = new AvroCSVReader<>(open(fs, inputPath), props, schema,
-        Record.class, false);
-    return new CsvReader(fs, reader, schema);
+    return new CsvReader(fs, inputPath);
```

Comment:
Can we do `new CsvReader(inputPath)` and do the conf/fs creation in the constructor?

Reply:
sure